event-svc/src/event/validator/event.rs

```diff
@@ -0,0 +1,151 @@
+use std::sync::Arc;
+
+use ceramic_core::{Cid, EventId};
+use ceramic_event::unvalidated;
+use ipld_core::ipld::Ipld;
+use recon::ReconItem;
+
+use crate::{
+    event::{service::InvalidItem, DeliverableRequirement},
+    store::{EventInsertable, SqlitePool},
+    Result,
+};
+
+#[derive(Debug)]
+pub struct ValidatedEvents {
+    /// These events are valid
+    pub valid: Vec<ValidatedEvent>,
+    /// We don't have enough information to validate these events yet.
+    /// e.g. The init event is required and the node has not yet discovered it.
+    pub pending: Vec<UnvalidatedEvent>,
+    /// Events that failed validation
+    pub invalid: Vec<InvalidItem>,
+}
+
+#[derive(Debug)]
+pub struct ValidatedEvent {
+    key: EventId,
+    cid: Cid,
+    event: Arc<unvalidated::Event<Ipld>>,
+}
+
+impl ValidatedEvent {
+    pub fn order_key(&self) -> &EventId {
+        &self.key
+    }
+}
+
+// TODO: should this impl from or have a private method?
+// it's not supposed to be public outside the crate and it currently allows us to
+// skip validation for the cases we're not doing it
+impl From<UnvalidatedEvent> for ValidatedEvent {
```

Comment:
yeah, that's reasonable. I think maybe this can be removed/just a private method now that validation accepts a "don't actually do validation" option (e.g. for migrations)